### DISCUSSION TITLE
Delegate valid checking to the compiler

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2837] [Suggestions]
1. Deduplicates a *lot* of code
2. Can optimize by caching (most likely per block, and would need destructive functions to invalidate the cache somehow (attribute? context method?))

**Comment by vurvdev** [comment]
Need more detail on what you mean here

**Reply by Denneisk** [reply]
https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/entity.lua#L105

```lua
if not IsValid(this) then return self:throw("Invalid entity!", "") end
```

These sorts of checks (similar can be found in EGP extensions, wirelink, etc.).

**Reply by thegrb93** [reply]
As long as there's a way to disable the check for cases where an invalid entity is allowable.
